### ISSUE TITLE
About the annotations of keypoints

### ISSUE THREAD

**Issue by buaaswf** [silviazuffi/smalst#12]
Dear @silviazuffi ,
Thanks for your sharing. Excellent works!
Could you please upload the code to generate keypoints of the input image?
I want to apply this to my customed dataset without keypoints. Could you please give me some suggestions of the 
keypoints generation?

Thanks very much.

**Comment by silviazuffi** [comment]
Hi, 
We do not use key points on the input image, we only use keypoints during training, and these are obtained from the 3D avatars.
You can just use the project_points function of the renderer to project selected 3D vertexes of the model on the image plane.

**Comment by buaaswf** [comment]
Dear @silviazuffi ,
Thanks for your quick reply. I will try the method you told me.
Besides, I have another question. Given a full image, how could I process it as the cropped one. What operations did you do
when you process the original image. I guess you handle these images with horse detection and segmentation, as well as align all the images. Could please upload the process code?
Thanks very much,
yours
Wenfeng

**Comment by silviazuffi** [comment]
We got the images from Wildbook (https://zebra.wildbook.org/ <https://zebra.wildbook.org/>) through our co-author, and those were already as bounding boxes.
I think you can run something like Mask R-CNN to identify the zebras

**Comment by buaaswf** [comment]
Yes, definitely I could get the boxes and mask via Mask R-CNN, however, I saw your zebra images are all aligned so that the images are in the same locations and even all the heads of the zebra are in the right side, I wonder how to process all the images to get such results. In addition, whether this processing results will have a influence on the mesh model generation.

**Comment by silviazuffi** [comment]
We mirror the images, so they are not all from the same side.

**Comment by buaaswf** [comment]
Yes, I get it. Thanks for your kind reply.

**Comment by buaaswf** [comment]
Dear @silviazuffi , 
I have sent you the results of my customer dataset, please check your emaill.silvia@mi.imati.cnr.it. Thanks very much,

**Comment by silviazuffi** [comment]
Are those images of Grevy’s zebras?

**Comment by buaaswf** [comment]
No， I apply it to some web searched horse images.

**Comment by silviazuffi** [comment]
The pre-trained network is trained on Grevy’s Zebras, it cannot be used for horses.

**Comment by buaaswf** [comment]
Hi @silviazuffi ,
What should I do to train a new model for horses?

**Comment by silviazuffi** [comment]
You would need to create a synthetic dataset for horses like the one for the zebras.
For doing this you need to have SMAL models of horses with variable textures and a 3D prior for sampling horse poses.

**Comment by buaaswf** [comment]
hi @silviazuffi , 
I have read your works '3D Menagerie: Modeling the 3D Shape and Pose of Animals', This is great work.
I saw some demos of reconstructing 3d mesh from the 2d images. Could you send me this part of the code to do further research?

**Comment by Autumnfall-Me** [comment]
hi @silviazuffi ,
I want to make myself SMAL model such as pig, but I don't know the model .pkl file.  Now I have a .obj model which contains the information of v, vt, f...Could you please tell me how to make the .pkl model which contains J,betas?

**Comment by silviazuffi** [comment]
Hi,
You should align the SMAL model to your 3D model.
If you look at this repository https://github.com/silviazuffi/smalr_online <https://github.com/silviazuffi/smalr_online> there are functions to compute mesh distances. You can use those to perform the alignment.

Best

**Comment by Autumnfall-Me** [comment]
Dear @silviazuffi ,
Thanks for solving my confusion. I will try the method you told me.

**Comment by cheimatrabelsi** [comment]
Dear Silivia, 
I read the Supplemental Material for the SMALST model, where you mentioned generating a synthetic training set and training a network for horses . Is the dataset published anywhere? If not,  can you share it with me? I'm interested in using it and comparing it to my current dataset Thank you in advance?

**Comment by silviazuffi** [comment]
Yes, I still have the dataset, it is about 9K images, I am quite busy in these days, but I will try to make it available.

**Comment by cheimatrabelsi** [comment]
Thank you for your reply. I truly appreciate your willingness to make the
dataset available
